### PR TITLE
Pay using dropDown button

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-45

### Description

-create pay using dropdown button 
-payment option is create 

### Screenshots
Pay using dropdown button
Simulator Screenshot - iPhone 16 Pro - 2024-12-10 at 17 48 16

### Tips
```bash
const phonepeurl =
    'https://uxwing.com/wp-content/themes/uxwing/download/brands-and-social-media/phonepe-icon.png';
const googlepayurl =
    'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJtedIIIMomjzOmz2R7pPBC0kGt7TwcIyWSw&s';
const amazonpayurl =
    'https://www.iconpacks.net/icons/free-icons-6/free-amazon-pay-circle-round-logo-icon-19775-thumb.png';
const paytmurl = 'https://thepatriot.in/wp-content/uploads/2024/02/Paytm.jpg';
final listOfPayusing = [
  PayUsingModel(phonepeurl, 'PhonePe', 'phonepe'),
  PayUsingModel(googlepayurl, 'Google Pay', 'tez'),
  PayUsingModel(amazonpayurl, 'Amazon Pay', 'amazonpay'),
  PayUsingModel(paytmurl, 'Paytm UI', 'tez'),
];

class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Scaffold(
      backgroundColor: theme.colors.light,
      body: Padding(
        padding: const EdgeInsets.all(40),
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.end,
          children: [
            Expanded(
              child: PayUsingButton(
                listOfPayusing: listOfPayusing,
              ),
            ),
            const Align(
              alignment: Alignment.bottomCenter,
              child: ActionButton(
                buttonType: ButtonType.text,
                title: 'Pay now',
                icon: Icons.abc_outlined,

              ),
            )
          ],
        ),
      ),
    );
  }
} 
```

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'main' branch
- [x] PR is targeting only one issue
- [ ] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin

### Primary reviewer checklist

- [ ] All items in the author checklist is checked or not applicable
- [ ] All files including .lock and assets are reviewed
- [ ] Reasonable UI flows/cases related to the change are manually tested
- [ ] No commented-out code or unnecessary debug statements.
- [ ] No spelling or grammatical errors in code comments or documentation.
- [ ] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [ ] The branch is up-to-date with the target branch (e.g., staging).

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: lib/pay_using_button.dart [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:iconsax/iconsax.dart';
+import 'theme/theme.dart';
+
+class PayUsingModel {
+  final String imageUrl;
+  final String providerName;
+  final String upiPrifix;
```
Comment (quiteachaos) [inline lib/pay_using_button.dart:9]: upiPrefix*

File: lib/pay_using_button.dart [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  PayUsingModel(
+    this.imageUrl,
+    this.providerName,
+    this.upiPrifix,
+  );
+}
+
+class PayUsingButton extends StatefulWidget {
+  final List<PayUsingModel> listOfPayusing;
```
Comment (quiteachaos) [inline lib/pay_using_button.dart:18]: listOfPayUsing

File: lib/pay_using_button.dart [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            ),
+            SizedBox(width: theme.spacing.width.s12),
+            Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                Row(
+                  children: [
+                    Text(
+                      'Pay Using',
```
Comment (quiteachaos) [inline lib/pay_using_button.dart:127]: use string_constants. payUsingText='Pay Using'

**Review by quiteachaos — DISMISSED** [review]

File: lib/pay_using_button.dart (hunk 1 above)
Comment (krushnakantanayak) [reply lib/pay_using_button.dart]: done

File: lib/pay_using_button.dart (hunk 2 above)
Comment (krushnakantanayak) [reply lib/pay_using_button.dart]: done

File: lib/pay_using_button.dart (hunk 3 above)
Comment (krushnakantanayak) [reply lib/pay_using_button.dart]: done

**Review by satyabratanayak — APPROVED** [review]

**Review by quiteachaos — APPROVED** [review]

**Review by satyabratanayak — APPROVED** [review]

**Pull request merged**